web/modules/weather_blocks/src/Plugin/Block/WeatherStoryBlock.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\weather_blocks\Plugin\Block;
+
+/**
+ * Provides a block of the active weather story for a WFO.
+ *
+ * @Block(
+ *   id = "weathergov_weather_story",
+ *   admin_label = @Translation("Weather story block"),
+ *   category = @Translation("weather.gov"),
+ * )
+ */
+class WeatherStoryBlock extends WeatherBlockBase
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function build()
+    {
+        $location = $this->getLocation();
+
+        if ($location->grid) {
+            $grid = $location->grid;
+
+            // Get the ID for the WFO taxonomy term that matches our grid WFO.
+            $termID = $this->entityTypeManager
+                ->getStorage("taxonomy_term")
+                ->loadByProperties(["field_wfo_code" => $grid->wfo]);
+
+            // If we don't get any results, that means we don't have a WFO
+            // taxonomy code for this WFO. By definition, we also can't have any
+            // weather stories for it, so we can bail out now.
+            if (count($termID) === 0) {
+                return null;
```

Comment:
I am not 100% sure about this, but I think Block `build()` methods need to always return an array, otherwise the whole application will crap out. In this case we would want an empty array. However, if we are trying to leave it as is for the future issue of dealing properly with error handling then this makes sense.

Reply:
Changed to `[]`. I think you're right. It was working but that may have been luck. 🤷‍♂️